packages/Webkul/User/src/Http/Middleware/Bouncer.php

```diff
@@ -84,4 +100,37 @@ public function checkIfAuthorized()
             bouncer()->allow($roles[Route::currentRouteName()]);
         }
     }
+
+    /**
+     * Check if two-factor authentication is required.
+     *
+     * @param  string  $guard
+     * @return bool
+     */
+    public function isTwoFactorRequired($guard)
+    {
+        $admin = auth()->guard($guard)->user();
+
+        // Only require 2FA if admin has it enabled AND hasn't verified in this session
+        if ($admin && $admin->two_factor_enabled && ! session()->get('two_factor_passed', false)) {
```

Comment:
Move the session logic in different method like `isTwoFactorPassed` so that we can add additional details specific to this.